### ISSUE TITLE
jsx-max-props-per-line - ignore tags with maximum attributes when maximum === 1

### ISSUE THREAD

**Issue by dylang** [yannickcr/eslint-plugin-react#1127]
## Currently

```
'react/jsx-max-props-per-line': ['error', { maximum: 1, when: 'always' }],
'react/jsx-closing-bracket-location': ['error', 'tag-aligned'],
```

### Before

```jsx
<Component styles={styles} />
````

### After

```jsx
<Component
    styles={styles} 
/>
````

## Expected

Because the tag above has 1 attribute, it should not need to be modified to put that attribute on a new line.

```jsx
<Component styles={styles} />
````

## Workaround

Limit to `multiline`:

```
'react/jsx-max-props-per-line': ['error', { maximum: 1, when: 'multiline' }]
```

However this means we cannot set the maximum number of attributes for single-line tags.

**Comment by ljharb** [comment]
I think it makes sense for `"when": "multiline"` to obey `maximum` - I'm surprised it does not.

If it's *not* going to obey it, it should be an early schema error to specify both.

**Comment by golopot** [comment]
Cannot reproduce this bug.

This test case covers it:
https://github.com/yannickcr/eslint-plugin-react/blob/master/tests/lib/rules/jsx-max-props-per-line.js#L31
 (this rule defaults to `{ maximum: 1, when: 'always' }` )

**Comment by ljharb** [comment]
Closing as fixed; happy to reopen if it's not.
